### PR TITLE
[IMP] orm,base,web,*: add falsy_value_label attr on fields

### PR DESCRIPTION
**Pull request by aab-odoo**
*hr_timesheet,project

This commit allows to specify, on a field definition, the value to display when that field is not set. That value is then used by the webclient in various views, when grouped, as display name of the group representing the `false` value (instead of "None", which is the default).

Previously, to achieve this, one had to define custom views (js_class) for each view type, and to override the adequate
function. The "logic" was thus replicated in all views where we wanted the customization. Now, it is defined at a single place, and automatically applied to all views (kanban, list, pivot, graph + map in enterprise).

This commit takes the benefit of the new attribute in project, where there were overrides for all these view types for the task model, to display "Private" when grouped by `project_id` and "Unnassigned" when grouped by `user_ids`. A lot of js_class have thus been removed.

This commit also factorizes the group displayName logic in kanban and list by moving it to the RelationalModel. The group datapoint already had a `displayName` property, but it wasn't computed correctly (it was not even always a string), so some additional processing was required in both list and kanban. We now apply this logic directly in the model, s.t. the displayName of a group is now always a string and can be used directly by the views.

Task~4410227

### CODE REVIEW

File: odoo/addons/base/models/ir_ui_view.py [hunk 1]
```diff
@@ -2565,9 +2565,11 @@ def get_views(self, views, options=None):
         result['models'] = {}
 
         for model, model_fields in models.items():
-            result['models'][model] = {"fields": self.env[model].fields_get(
-                allfields=model_fields, attributes=self._get_view_field_attributes()
-            )}
+            result['models'][model] = {
```
Comment (Polymorphe57) [inline odoo/addons/base/models/ir_ui_view.py:2568]: remove diff

**Review by Polymorphe57 — APPROVED** [review]

**Review by kmagusiak — APPROVED** [review]
python changes look ok

**Review by xavierbol — APPROVED** [review]
LGTM for the services part, thanks for your work! 🤩

**Comment by Polymorphe57** [comment]
@robodoo r+

**Review by rco-odoo — APPROVED** [review]
Good to me.

**Comment by aab-odoo** [comment]
robodoo r+

**Pull request closed without merging**